src/modules/commander/state_machine_helper.cpp

```diff
@@ -616,8 +616,8 @@ bool set_nav_state(vehicle_status_s *status, actuator_armed_s *armed, commander_
 			// is not possible and therefore the internal_state needs to be adjusted.
 			internal_state->main_state = commander_state_s::MAIN_STATE_POSCTL;
 
-		} else if (rc_lost && !data_link_loss_act_configured && is_armed) {
-			// failsafe: RC is lost, datalink loss is not set up and rc loss is not disabled
+		} else if (rc_lost && status->data_link_lost && !data_link_loss_act_configured && is_armed) {
+			// Orbit does not depend on RC but while armed & all links lost even when datalink loss is not set up we failsafe
```

Comment:
Sorry to be so picky, but I understand your comment as:
`rc_lost && status->data_link_lost && is_armed` as "even when datalink loss is not set up" means that the value of `data_link_loss_act_configured` is not considered.

Reply:
"			// Orbit does not depend on RC but while armed & all links lost & when datalink loss is not set up, we failsafe"

Reply:
I like picky reviews 👍 
You're right, that's more precise. My comment contained the context of the previous case handling the case when data link loss is configured.

We should probably revise this part of redundant state machine logic with the bottom line it will failsafe if you have rc loss configured and lose all links. The only difference between the cases is which action it triggers. Is there a reason for that? That's a new topic.

Reply:
I applied your suggestion, thanks: https://github.com/PX4/PX4-Autopilot/compare/d12af88ba2e8781caa6a29560a0e8a292ea3f0d8..f98a5c6cdeaa2272fba9986adeb924797e44c524